### PR TITLE
Np.inf

### LINKED ISSUE #1285: ml.py contains conflicting references to Numpy
flaml/automl/ml.py

On line 249, you have: "return np.inf" with lower case i but on line 326 you have "best_val_loss=np.Inf" with upper case I. 

The latter one is deprecated in Numpy 2.0 and causes an error with e.g. Autogen (or pyautogen): 

**AttributeError: `np.Inf` was removed in the NumPy 2.0 release. Use `np.inf` instead.. Did you mean: 'inf'?** 

  File "<path to lib>\Lib\site-packages\flaml\automl\automl.py", line 15, in <module>
    from flaml.automl.state import SearchState, AutoMLState
Line 326, in <module>
    best_val_loss=np.Inf,

### PR DESCRIPTION
**Pull request by sonichi**
## Why are these changes needed?

np.Inf is deprecated in numpy v2.

## Related issue number

Closes #1285

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Review by davorrunje — APPROVED** [review]
LGTM

File: flaml/automl/automl.py [hunk 1]
```diff
@@ -2647,7 +2647,7 @@ def _select_estimator(self, estimator_list):
             if self._estimator_index == len(estimator_list):
                 self._estimator_index = 0
             return estimator_list[self._estimator_index]
-        min_estimated_cost, selected = np.Inf, None
+        min_estimated_cost, selected = np.inf, None
```
Comment (Borda) [inline flaml/automl/automl.py:2650]: What is the min `numpy` which enables this?

Comment (sonichi) [reply flaml/automl/automl.py:2650]: flaml requires numpy>=1.17. 1.17 has it.

**Comment by jackgerrits** [comment]
From the 2.0 release notes:
- Alias np.Inf has been removed. Use np.inf instead.

https://numpy.org/devdocs/release/2.0.0-notes.html

**Review by jackgerrits — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Pull request merged**